### PR TITLE
planner: allow insert `default` into generated columns

### PR DESCRIPTION
**Pull request by Deardrops**
### What problem does this PR solve? 
[In MySQL document](https://dev.mysql.com/doc/refman/5.7/en/create-table-generated-columns.html), 
> For INSERT, REPLACE, and UPDATE, if a generated column is inserted into, replaced, or updated explicitly, the only permitted value is DEFAULT.

```sql
create table t (a int, b int as (a + 1))
insert into t value (1, default)
MySQL: Query OK (0.191 sec)
TiDB: Error 3105: The value specified for generated column 'b' in table 't' is not allowed.
```
### What is changed and how it works?

Allow use the `DEFAULT` keyword when insert into the generated columns.

When build the `insert plan`, check if
- the target column is generated column, 
- and the insert expression is `DEFAULT`

Skip add this column's expression into this `insert plan`. 

Because we can't insert value into generated clumns directly, it's acceptable to ignore the generated column in `insert plan`.

### Check List 

Tests 

 - Intergration test

Release note

 - Allow use the `DEFAULT` keyword when insert into the generated columns.

### CODE REVIEW

File: planner/core/planbuilder.go [hunk 1]
```diff
@@ -1778,8 +1772,14 @@ func (b *PlanBuilder) buildValuesListOfInsert(ctx context.Context, insert *ast.I
 		for j, valueItem := range valuesItem {
 			var expr expression.Expression
 			var err error
+			var generatedColumnWithDefaultExpr bool
+			col := affectedValuesCols[j]
 			switch x := valueItem.(type) {
 			case *ast.DefaultExpr:
+				if col.IsGenerated() {
```
Comment (wjhuang2016) [inline planner/core/planbuilder.go]: Need to check if the DefaultExpr is only `default`.
```
mysql> insert into t value (default,  default(a));
Query OK, 1 row affected (0.01 sec)
```
However, in MySQL:
```
mysql> insert into t value (default,  default(a));
ERROR 3773 (HY000): DEFAULT function cannot be used with default value expressions
```

Comment (Deardrops) [reply planner/core/planbuilder.go]: @wjhuang2016 Comment Addressed

**Comment by wjhuang2016** [comment]
LGTM

**Comment by tangenta** [comment]
@Deardrops Do we have the plan to support `insert into t set a = 1, b = default`?

**Comment by Deardrops** [comment]
@tangenta 
> @Deardrops Do we have the plan to support `insert into t set a = 1, b = default`?

Yes, we have open a issue for this problem https://github.com/pingcap/tidb/issues/11939

**Pull request merged**